### PR TITLE
Added error checking, extracted token and connetion builders to use t…

### PR DESCRIPTION
**Pull request by luis-falcon**
Added error checking, extracted token and connetion builders to use the utils function, all support commands can use any cluster identifyer, etc

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -19,3 +28,125 @@ func GetOCMAccessToken() (*string, error) {
...
+		Send()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get limited Support Reasons: %s", err)
+	}
+
+	lmtReason := limitedSupportReasons.Items()
+
+	var clusterLmtSprReasons []*lmtSprReasonItem
+
+	lmtReason.Each(func(lmtSprReason *cmv1.LimitedSupportReason) bool {
```
Comment (fahlmant) [inline pkg/utils/utils.go:140]: Everything looks good here, just curious what this Each is doing?

Comment (fahlmant) [reply pkg/utils/utils.go:140]: Why was this resolved?

Comment (fahlmant) [inline pkg/utils/utils.go:140]: What is this each doing?

Comment (luis-falcon) [reply pkg/utils/utils.go:140]: you asked a question. I answered, what else was there to do to? Am I missing something here? is that not how it is supposed to be done? Or is the Conversation starting party the one that should be "resolving"?  :thinking:

Comment (fahlmant) [reply pkg/utils/utils.go:140]: It was resolved before you actually posed the response :) 

Screenshot from 2022-07-13 11-41-46

Comment (luis-falcon) [reply pkg/utils/utils.go:140]: ahhh. Ok cool :/ is good to know

Comment (luis-falcon) [reply pkg/utils/utils.go:140]: Hey @fahlmant this is because we get a list when we get the limited support reasons. This is because a cluster can have many reasons why it is on limited support. Thus, for each of those reasons (1 or many) on that limited support list, we create a reason object and append them to the slice we will return. We do this Right here instead of let's say just return the full list to the code calling this function because we also want to get only the pertinent properties (the ones we care about) for the limited support reasons and because that way if we need this code at some point we don't need to parse that list, we just get the slice of reasons we need and job done. And if we want more or less at some point we can modify this struct and the .each part of the code.

So In summary, we looping through each limited support reason and extracting only the info we need.

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -19,3 +28,125 @@ func GetOCMAccessToken() (*string, error) {
...
+		clusterLmtSprReason := lmtSprReasonItem{
+			ID:      lmtSprReason.ID(),
+			Summary: lmtSprReason.Summary(),
+			Details: lmtSprReason.Details(),
+		}
+		clusterLmtSprReasons = append(clusterLmtSprReasons, &clusterLmtSprReason)
+
+		return true
+	})
```
Comment (fahlmant) [inline pkg/utils/utils.go:149]:
```suggestion
	for _, resp := range lmtReason {
		clusterLmtSprReason := lmtSprReasonItem{
			ID:      resp.ID(),
			Summary: resp.Summary(),
			Details: resp.Details(),
		}
		clusterLmtSprReasons = append(clusterLmtSprReasons, &clusterLmtSprReason)
	}
```

I think putting this directly in a loop is better than using an anonymous function if we're only ever going to return true

Comment (luis-falcon) [reply pkg/utils/utils.go:149]: I tried like this, this will not work, the range function is trying then to iterate through a pointer which is what the lmtReason is.

Comment (fahlmant) [reply pkg/utils/utils.go:149]: Looks like you can use `UnmarshalLimitedSupportReasonList` to get a list of limited support reasons, and then iterate through that. 

 https://github.com/openshift-online/ocm-sdk-go/blob/main/clustersmgmt/v1/limited_support_reason_list_type_json.go#L56

**Review by fahlmant — APPROVED** [review]
/approve
/lgtm
